synapse/handlers/auth.py

```diff
@@ -863,11 +863,15 @@ async def delete_access_token(self, access_token: str):
         # see if any of our auth providers want to know about this
         for provider in self.password_providers:
             if hasattr(provider, "on_logged_out"):
-                await provider.on_logged_out(
+                # This might return an awaitable, if it does block the log out
+                # until it completes.
+                result = provider.on_logged_out(
                     user_id=str(user_info["user"]),
                     device_id=user_info["device_id"],
                     access_token=access_token,
                 )
+                if result:
```

Comment:
```suggestion
                if isinstance(result, Awaitable):
```

perhaps? Though there's a question about whether we want things like this to fail loudly rather than quietly.

You would need to `from typing import Awaitable` as well of course.

Reply:
In other places we've been using `inspect.isawaitable`, I suppose this is actually more compatible with the old code (which was a `yield` call).

Note that this doesn't affect either of the known password providers, which both return a `Deferred`.

Reply:
Done in e99f914803582b52a0d53a874702fc274c48eddf!